rust/connlib/snownet/src/node.rs

```diff
@@ -1261,9 +1204,21 @@ struct InitialConnection {
 
     created_at: Instant,
     intent_sent_at: Instant,
+
+    is_failed: bool,
```

Comment:
It is stored because I need to remember that the connection is failed after handling a timeout event (from wireguard or ICE) to clean-up the connection afterwards. It is not ideal.

The problem is, whilst I am calling `handle_timeout` on the connection, I cannot also remove it because I am holding a mutable reference to it. Hence, I need the connection it change its internal state and later I clean them up using `retain` on the `HashMap`.

I am very much open to a different design here. This probably also deserves a comment.